### PR TITLE
3814 - Event content model

### PR DESCRIPTION
**Pull request by briaguya**
https://github.com/cityofaustin/techstack/issues/3816
https://github.com/cityofaustin/techstack/issues/3814

# Description

* Adds event pages to Joplin

# Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How can this be tested?

 
https://joplin-pr-3814-events-non-joyo.herokuapp.com/

# Checklist:
- [x] Request reviewers
- [ ] Slack-out message for review
- [x] Link this PR in the issue
- [x] Moved card to *review* in zenhub

### CODE REVIEW

File: joplin/api/schema.py [hunk 1]
```diff
@@ -281,7 +393,7 @@ class Language(graphene.Enum):
     BURMESE = 'my'
 
 
-class ServicePageStepLocationBlock(graphene.ObjectType):
+class StreamFieldLocationPage(graphene.ObjectType):
```
Comment (easherma) [inline joplin/api/schema.py]: this is a name change unrelated the event content type, no? I feel like the old name made sense. Does this need to be a part of this PR?

File: joplin/api/schema.py [hunk 2]
```diff
@@ -193,6 +194,117 @@ class Meta:
         model = LocationPageRelatedServices
         interfaces = [graphene.Node]
 
+class EventPageRemoteLocation(graphene.ObjectType):
```
Comment (easherma) [inline joplin/api/schema.py]: I remember we spoke about this verbally, but I think a docstring or extended comment here would be helpful, to explain why we aren't using the DjangoObjectType but wrote a 'custom schema'.

Also see my note below about the name of the data model

File: joplin/api/schema.py [hunk 3]
```diff
@@ -193,6 +194,117 @@ class Meta:
...
+        return self.value['country']
+
+    def resolve_zip(self, info):
+        return self.value['zip']
+
+    name = graphene.String()
+
+    def resolve_name(self, info):
+        # We're doing our own translations in our model here
+        # so let's make sure the API still works as expected
```
Comment (easherma) [inline joplin/api/schema.py]: would be great to clarify for future readers what 'as expected' means

File: joplin/events/models.py [hunk 4]
```diff
@@ -0,0 +1,157 @@
...
+            ('city_location', StructBlock(
+                [
+                    ('location_page', PageChooserBlock(label="Location", page_type=[LocationPage], classname='do-not-hide')),
+                    ('additional_details_en', TextBlock(label='Any other necessary location details, such as room number [en]', required=False)),
+                    ('additional_details_es', TextBlock(label='Any other necessary location details, such as room number [es]', required=False)),
+                    ('additional_details_ar', TextBlock(label='Any other necessary location details, such as room number [ar]', required=False)),
+                    ('additional_details_vi', TextBlock(label='Any other necessary location details, such as room number [vi]', required=False)),
+                ]
+            )),
+            ('remote_location', StructBlock(
```
Comment (easherma) [inline joplin/events/models.py]: I realize this is a minor quibble, but I think the name 'remote location' could be kinda confusing. It makes me think that the event could be a webinar or somehow not in person. 

I know content side its described as 'non-city location', so I'd expect it to be called something like 'non_city_location', 'NonCityLocation', etc. It'l make it much easier to find in the codebase in the future.

File: joplin/js/editor.js [hunk 5]
```diff
@@ -223,16 +224,20 @@ $(function() {
...
-              fieldlabel.parentNode.classList.remove('hidden')
-              if (attrFor[attrFor.length-1] !== currentLang) {
-                fieldlabel.parentNode.classList.add('hidden')
-                translatedElement.classList.remove('hidden') // only re-reveal the parent class if we find this case.
+          if (translatedElement.classList.contains('struct-block')) {
+            const fieldlabels = translatedElement.querySelectorAll('[for]');
+            fieldlabels.forEach(fieldlabel => {
+              const attrFor = fieldlabel.getAttribute('for').split('_');
+              fieldlabel.parentNode.classList.remove('hidden');
+              // Adding a failsafe to make sure we don't remove non translated fields
```
Comment (easherma) [inline joplin/js/editor.js]: * party emoji *

File: joplin/api/schema.py.orig [hunk 6]
```diff
@@ -0,0 +1,954 @@
+import django.utils.translation
+import graphene
```
Comment (chiaberry) [inline joplin/api/schema.py.orig]: do we need to keep the old schema? there are merge conflicts to boot

Comment (easherma) [reply joplin/api/schema.py.orig]: Wait, yeah I totally missed this. @briaguya whats the deal with this file?

File: joplin/api/schema.py (hunk 1 above)
Comment (briaguya) [reply joplin/api/schema.py]: I was going to try to reuse it for city_location on event page but forgot it was actually building an array, forgot to undo the rename when I decided not to go that route. I'm fixing it rn.

File: joplin/events/models.py (hunk 4 above)
Comment (briaguya) [reply joplin/events/models.py]: @desigonz and I discussed and the plan is to have content come up with a final name, but to use remote location until we have a decision from them. I'll add some comments to the code.

File: joplin/api/schema.py (hunk 3 above)
Comment (briaguya) [reply joplin/api/schema.py]: addressed in new comments

File: joplin/api/schema.py (hunk 2 above)
Comment (briaguya) [reply joplin/api/schema.py]: addressed in new comments

File: joplin/api/schema.py.orig (hunk 6 above)
Comment (briaguya) [reply joplin/api/schema.py.orig]: whoops didn't mean to check that in, good catch (too much merging)

**Pull request merged**